### DISCUSSION TITLE
404 File not found

### DISCUSSION THREAD

**Discussion by kenturkey048** [community/community#126831] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

i am trying to host my repos on git up pages.. I have been getting this error..Don't know what the problem is..
please somebody should  help me..
 
{{ 404
File not found

The site configured at this address does not contain the requested file.

If this is your site, make sure that the filename case matches the URL as well as any file permissions.
For root URLs (like http://example.com/) you must provide an index.html file.

[Read the full documentation](https://help.github.com/pages/) for more information about using GitHub Pages }}

**Comment by davevad93** [comment]
Hi @kenturkey048 ,

GitHub Pages looks for the `index.html` file in the root of your repository, so make sure your `html` is there.

**Reply by kenturkey048** [reply]
hi.. Thanks for your response.. To my understanding i think i did everything rightly
..
below is the repos code.. Please can you go through it?

https://kenturkey048.github.io/Space-Xproject/

**Reply by kenturkey048** [reply]
sorry .. This is the link.. The first was a mistake
https://github.com/kenturkey048/Space-Xproject

**Reply by davevad93** [reply, marked as answer]
Hey @kenturkey048 , sorry for the late reply I've been busy. I've taken a look at your repo and the thing is your `index.html` isn't in the root of the repo, but it's inside the `public` folder, the structure should be as follows:

/Space-Xproject
│
├── public
├── src
├── .gitignore
├── index.html
├──package-lock.json
├── package.json
└── tailwind.css

Note that you will probably have to update the path of your `css`, `js` and `img` files within the `index.html` file.

**Reply by kenturkey048** [reply]
Thank you so much for your response.. With your guidance, i was able to get up and running...

**Reply by davevad93** [reply]
Hey @kenturkey048 you're welcome, happy to help!

**Reply by davevad93** [reply]
Hi @hfksue ,

I noticed that your project is built with `vue` and it's a bit different to configure, I've never deployed a project using `vue` so I really don't know how to help you properly here. Considering you need an `index.html` file in the repo root you could try putting the file there leaving the one in the `public` folder as well but maybe the problem is with overall setup so check out these links on creating a `vue.js` project on GitHub Pages:

- [How To Deploy Your Vue App to GitHub Pages](https://learnvue.co/articles/deploy-vue-to-github-pages) (article)
- [How to Deploy Your Vite App to Github Pages](https://www.youtube.com/watch?v=yo2bMGnIKE8&t=10s) (Youtube video)

**Reply by LiteBrite82** [reply]
also, be sure to add a `.nojekyll` file to the root of your main branch. (just a file named `.nojelyll`; nothing needs to be in it) [Guide: Troubleshooting 404 errors for GitHub Pages sites](https://github.com/orgs/community/discussions/61073)
